app/src/panels/time-series/panel-time-series.vue

```diff
@@ -6,27 +6,36 @@
 
 <script setup lang="ts">
 import { useFieldsStore } from '@/stores/fields';
-import { Filter } from '@directus/types';
+import type { Filter } from '@directus/types';
 import { abbreviateNumber, adjustDate } from '@directus/utils';
 import { cssVar } from '@directus/utils/browser';
 import ApexCharts from 'apexcharts';
 import { addWeeks } from 'date-fns';
-import { isNil, snakeCase } from 'lodash';
+import { isNil, orderBy, snakeCase } from 'lodash';
 import { computed, onMounted, onUnmounted, ref, watch } from 'vue';
 import { useI18n } from 'vue-i18n';
-import { orderBy } from 'lodash';
 
 const props = withDefaults(
 	defineProps<{
 		height: number;
 		showHeader?: boolean;
-		data?: object[];
+		data?: {
+			group: Record<string, number | string>;
+			count: Record<string, number>;
+			countDistinct: Record<string, number>;
+			avg: Record<string, number>;
+			avgDistinct: Record<string, number>;
+			sum: Record<string, number>;
+			sumDistinct: Record<string, number>;
+			min: Record<string, number>;
+			max: Record<string, number>;
+		}[];
 		id: string;
 		now: Date;
 		collection: string;
 		dateField: string;
 		valueField: string;
-		function: string;
+		function: 'count' | 'countDistinct' | 'avg' | 'avgDistinct' | 'sum' | 'sumDistinct' | 'min' | 'max';
```

Comment:
Haven't we defined these types anywhere yet? And if not, wouldn't it be a good idea to do so?

Reply:
Apparently not, and yes. This panel already contains a ton of references to this full list as-is though, so lets worry about optimizations outside of the error fixes 👍🏻

Reply:
Sounds good 👍